synapse/storage/data_stores/main/events_bg_updates.py

```diff
@@ -503,3 +508,60 @@ def _event_fix_redactions_bytes_txn(txn):
         yield self._end_background_update("event_fix_redactions_bytes")
 
         return 1
+
+    @defer.inlineCallbacks
+    def _event_store_labels(self, progress, batch_size):
+        """Stores labels for events."""
+        last_event_id = progress.get("last_event_id", "")
+
+        def _event_store_labels_txn(txn):
+            txn.execute(
+                """
+                SELECT event_id, json FROM event_json
+                LEFT JOIN event_labels USING (event_id)
+                WHERE event_id > ? AND label IS NULL
+                ORDER BY event_id LIMIT ?
+                """,
+                (last_event_id, batch_size),
+            )
+
+            rows = self.cursor_to_dict(txn)
+            if not rows:
+                return True, 0
+
+            for row in rows:
+                event_id = row["event_id"]
+                event_json = json.loads(row["json"])
+
+                self._simple_insert_many_txn(
+                    txn=txn,
+                    table="event_labels",
+                    values=[
+                        {
+                            "event_id": event_id,
+                            "label": label,
+                            "room_id": event_json["room_id"],
+                            "topological_ordering": event_json["depth"],
+                        }
+                        for label in event_json["content"].get(
```

Comment:
this will blow up if someone sends events with non-string labels. please could you apply some more checks.